web/packages/teleport/src/Integrations/IntegrationList.tsx

```diff
@@ -124,6 +127,24 @@ export function IntegrationList(props: Props<IntegrationLike>) {
                         View Status
                       </MenuItem>
                     )}
+                    {item.kind === 'msteams' && (
+                      <MenuItem onClick={() => function () {
+			  api.fetch(cfg.getMsTeamsAppZipRoute(clusterId, item.name))
```

Comment:
You don't need to create an anonymous function and invoke it directly, the `() =>` is already an anonymous function. something like this
```js
onClick={() => {
    // api.fetch(blablba...you get it
}}
```

also, we have a `saveOnDisk` method that you should use here instead. [take a look](https://github.com/gravitational/teleport/blob/master/web/packages/shared/components/TextSelectCopy/TextSelectCopyMulti.tsx#L112) at an example of its use.

Reply:
Is something like this what you mean?

Reply:
yeah, perfect!

Reply:
the linter is throwing an error because `saveOnDisk` doesnt take a blob type. 
also, we don't have any error handling for this API call and just a couple formatting changes. Check out this diff and it should get you most of the way there (use with copy then `pbpaste | git apply`)

<details><summary>open me!</summary>

```diff
diff --git a/web/packages/shared/utils/saveOnDisk.ts b/web/packages/shared/utils/saveOnDisk.ts
index e201d368a2..6d61ec66a9 100644
--- a/web/packages/shared/utils/saveOnDisk.ts
+++ b/web/packages/shared/utils/saveOnDisk.ts
@@ -18,17 +18,18 @@
 
 /**
  * saveOnDisk saves content to local disk.
- * @param content content to download.
+ * @param content content to download (string or Blob).
  * @param filename preset file name.
  * @param fileType file type.
  */
 export function saveOnDisk(
-  content: string,
+  content: string | Blob,
   filename: string,
   fileType: string
 ): void {
   const a = document.createElement('a');
-  const blob = new Blob([content], { type: fileType });
+  const blob =
+    content instanceof Blob ? content : new Blob([content], { type: fileType });
   a.href = window.URL.createObjectURL(blob);
   a.download = filename;
   document.body.appendChild(a);
diff --git a/web/packages/teleport/src/Integrations/IntegrationList.tsx b/web/packages/teleport/src/Integrations/IntegrationList.tsx
index 12dc984643..56c8b32659 100644
--- a/web/packages/teleport/src/Integrations/IntegrationList.tsx
+++ b/web/packages/teleport/src/Integrations/IntegrationList.tsx
@@ -25,10 +25,12 @@ import { Box, Flex } from 'design';
 import Table, { Cell } from 'design/DataTable';
 import { MenuButton, MenuItem } from 'shared/components/MenuAction';
 import { ToolTipInfo } from 'shared/components/ToolTip';
+import { useAsync } from 'shared/hooks/useAsync';
 import { ResourceIcon } from 'design/ResourceIcon';
+import { saveOnDisk } from 'shared/utils/saveOnDisk';
+
 import useStickyClusterId from 'teleport/useStickyClusterId';
 import api from 'teleport/services/api';
-import { saveOnDisk } from 'shared/utils/saveOnDisk';
 
 import {
   getStatusCodeDescription,
@@ -68,8 +70,24 @@ export function IntegrationList(props: Props<IntegrationLike>) {
     return { cursor: 'pointer' };
   }
 
+  const [downloadAttempt, download] = useAsync(
+    async (clusterId: string, itemName: string) => {
+      return api
+        .fetch(cfg.getMsTeamsAppZipRoute(clusterId, itemName))
+        .then(response => response.blob())
+        .then(blob => {
+          saveOnDisk(blob, 'app.zip', 'application/zip');
+        });
+    }
+  );
+
   const { clusterId } = useStickyClusterId();
   return (
+    // render some error component somewhere like
+    // <>
+    //  {downloadAttempt.status === "error" && <Danger>{downloadAttempt.statusText}</Danger>}
+    // </>
+    //
     <Table
       pagination={{ pageSize: 20 }}
       isSearchable
@@ -116,15 +134,12 @@ export function IntegrationList(props: Props<IntegrationLike>) {
                       </MenuItem>
                     )}
                     {item.kind === 'msteams' && (
-                      <MenuItem onClick={() => {
-			  api.fetch(cfg.getMsTeamsAppZipRoute(clusterId, item.name))
-			      .then(response => response.blob())
-			      .then(blob => {
-				  saveOnDisk(blob, "app.zip", "application/zip")
-			      });
-		      }}>
+                      <MenuItem
+                        disabled={downloadAttempt.status === 'processing'}
+                        onClick={() => download(clusterId, item.name)}
+                      >
                         Download app.zip
-		      </MenuItem>
+                      </MenuItem>
                     )}
                     <MenuItem onClick={() => props.onDeletePlugin(item)}>
                       Delete...

```
</details>

Reply:
It works perfectly thanks for the help